server/region_cache.go

```
@@ -297,3 +297,70 @@ func (c *fifoCache) len() int {
 
 	return c.ll.Len()
 }
+
+type hashCache struct {
+	sync.RWMutex
+
+	maxCount int
+	length   int
+	items    map[interface{}]interface{}
+}
+
+func newHashCache(maxCount int) *hashCache {
+	return &hashCache{
+		maxCount: maxCount,
+		items:    make(map[interface{}]interface{}),
+	}
+}
+
+func (q *hashCache) IsFull() bool {
+	q.RLock()
+	defer q.RUnlock()
+	return q.isFull()
+}
+
+func (q *hashCache) isFull() bool {
+	return q.length >= q.maxCount
+}
+
+func (q *hashCache) Put(key interface{}, value interface{}) {
+	q.Lock()
+	defer q.Unlock()
+	if q.isFull() {
+		log.Info("Debug queue is full")
```

better to return a full error. 
Btw, what's Debug mean here?